modules/grizzly/src/main/java/org/glassfish/grizzly/ssl/SSLConnectionContext.java

```diff
@@ -361,7 +361,6 @@ SslResult wrap(final Buffer input, Buffer output,
         
         if (allocator != null && isOverflow) {
             updateBufferSizes();
-            output = ensureBufferSize(output, netBufferSize, allocator);
```

Comment:
I believe this is actually required. Should the output buffer be too small, the allocator will be null for the second call (which it shouldn't be going by the assertions).